inc/namespace.php

```diff
@@ -289,6 +294,12 @@ function ajax_query_attachments() : void {
 	if ( $post_id && ! current_user_can( 'edit_post', $post_id ) ) {
 		wp_send_json_error();
 	}
+	
+	// If no provider is specified and several posts are being requested,
+	// call the original AJAX handler, it's probably a gallery.
+	if ( empty( $args['provider'] ) && !empty( $args['post__in'] ) ) {
```

Comment:
```suggestion
	if ( empty( $args['provider'] ) && ! empty( $args['post__in'] ) ) {
```

Reply:
in practice, with local media disabled via the constant the custom DAM provider is what gets used as it's the _only_ provider, resulting in the first 10 images in that provider being returned as the contents of the gallery.

I'll investigate if this could be handled via our custom provider

Reply:
I see. Can you remind me of the workflow - you add images to the gallery, that will create the local stub attachment records, then when showing / editing the gallery it should return the localised not external media, is that right?

I _think_ this is ok but it'd be great if there was a way to label the request somehow as being for a gallery. There are ways to use jQuery middlewares to modify AJAX requests but might not be worth it. I'm trying to think if there actually is a legit use case where you would know the external media IDs ahead of time and need to use those...

There are additional parameters, `post__not_in` and `post_parent` that may be only pertinent to localised media too. It seems like your basic PR is sound but I would do the following:

- remove the check to see if the provider is empty in the if statement
- check if `post__not_in` and `post_parent` are also being used
- provide a way for providers to override/bypass that check and handle those parameters, could just be a property that can be overridden
- document it on the readme, and how to fallback to regular WP handling if you need to

@johnbillion would appreciate your thoughts here too.

Reply:
For the moment I've distilled a workaround that intercepts and handles the ajax action like this to fix galleries:

```php

remove_action( 'wp_ajax_query-attachments', 'AssetManagerFramework\\ajax_query_attachments' );
add_action( 'wp_ajax_query-attachments', __NAMESPACE__ . '\\replacement_ajax_query_attachments', 1 );

function replacement_ajax_query_attachments() : void {

	// phpcs:ignore WordPress.Security.NonceVerification.Recommended
	$args = isset( $_REQUEST['query'] ) ? wp_unslash( (array) $_REQUEST['query'] ) : [];

	// If no provider is specified and WP attachment IDs are referenced,
	// call the original AJAX handler, it's probably a gallery.
	if (
		empty( $args['provider'] ) &&
		(
			isset( $args['post__in'] ) ||
			isset( $args['post__not_in'] ) ||
			isset( $args['post_parent'] )
		)
	) {
		wp_ajax_query_attachments();
	}

	\AssetManagerFramework\ajax_query_attachments();
}
```